### PR TITLE
Added replication of events for bags created using the service.

### LINKED ISSUE #1: Events are working
The event replication isn't working for me.

The following prints the element and container properly to my console:
```
    dragulaService.options($scope, 'drag', {
      containers: Array.prototype.slice.call(document.querySelectorAll('ul.schedule-list'))
    });

    dragulaService.find($scope, 'drag').drake.on('drop', function(el, drop, source) {
      console.log(el, 'el');
      console.log(drop, 'drop');
    });
```

While this does nothing:
```
    dragulaService.options($scope, 'drag', {
      containers: Array.prototype.slice.call(document.querySelectorAll('ul.schedule-list'))
    });

    $scope.$on('drag.drop', function(el, drop, source) {
      console.log(el, 'el');
      console.log(drop, 'drop');
    });
```

### PR DESCRIPTION
**Pull request by kevinob11**
This is to resolve #1 and #3. This adds event replication to bags created by the service. The passing around of angular through each function call seems pretty wrong, I haven't used browserify for modules much is there a more elegant way of handling that?

### CODE REVIEW

**Comment by bevacqua** [comment]
Looks great but we need to remove the `replicateEvents` line from the directive now, since the service takes care of that

**Comment by bevacqua** [comment]
Similarly the directive should pass `angular` in the fourth argument as well.

Context: https://github.com/bevacqua/angular-dragula/blob/master/angular-dragula.js#L31-L32

**Comment by bevacqua** [comment]
Actually we should have the service receive `angular` so that we don't need to pass the reference around. I can merge and make that change if you want

**Comment by kevinob11** [comment]
Sure, that is actually what I was about to suggest.

**Comment by bevacqua** [comment]
You can do it if you want. I'm working on something else :ok_hand:

**Comment by kevinob11** [comment]
Sure, heading out for a bit but I can probably do it tonight.

**Comment by kevinob11** [comment]
Ok I did a bit more than I wanted to initially. The changes:
1. Events are replicated regardless of how you create the bag now, as we discussed
2. The directive now allows you to pass in the scope you want to use by passing that scope to `dragula-scope` on the element with the `dragula` directive
3. I moved the directive into its own file
4. I wrapped the directive and service in register functions so angular can be passed into those closures - I hate this but it appears to be the browserify / bower world we live in right now

Let me know what you think or if any other changes are needed.

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular-dragula.js",
-  "version": "1.0.3",
+  "version": "1.0.4",
```
Comment (bevacqua) [inline bower.json]: I'll take care of versions _(`npm run deploy` handles bumping version numbers)_

File: directive.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+var dragula = require('dragula');
+
+function register(angular) {
+	return function angularDragula (dragulaService) {
```
Comment (bevacqua) [inline directive.js]: You're missing a space here!

File: angular-dragula.js [hunk 3]
```diff
@@ -2,36 +2,15 @@
...
-var replicateEvents = require('./replicate-events');
+var dragulaDirective = require('./directive');
 
 function register (angular) {
   var app = angular.module('dragula', ['ng']);
 
-  app.factory('dragulaService', dragulaService);
-  app.directive('dragula', ['dragulaService', angularDragula]);
+  app.factory('dragulaService', dragulaService(angular));
+  app.directive('dragula', ['dragulaService', dragulaDirective(angular)]);
```
Comment (bevacqua) [inline angular-dragula.js]: Please move the entire array to the directive, as there's no reason why this module should know the dependencies of the other one, nor why we would want to have to change the dependencies in two files.

**Comment by bevacqua** [comment]
Looks good. Can you check the inline comments? Also, throw in some documentation in the README (or I can do that if you want)

File: bower.json (hunk 1 above)
Comment (kevinob11) [inline bower.json]: I think this happened when I rebased to your master, but I'm happy to change it back.

**Comment by kevinob11** [comment]
Ok, I think I've got everything fixed.

File: bower.json (hunk 1 above)
Comment (bevacqua) [inline bower.json]: Oops. You're right!

**Comment by bevacqua** [comment]
Great! I'll make a few tweaks and publish soon

**Comment by kevinob11** [comment]
Awesome, thanks!

**Comment by bevacqua** [comment]
Deployed as `1.0.5`

**Comment by kevinob11** [comment]
Great! I think you can close #3 now.

**Pull request merged**